packages/analytics-js/src/services/StoreManager/StoreManager.ts

```diff
@@ -96,7 +126,12 @@ class StoreManager implements IStoreManager {
       const key = sessionKey;
       const storageKey = sessionKey;
       const providedStorageType = entries?.[key]?.type;
-      const storageType = providedStorageType || globalStorageType || DEFAULT_STORAGE_TYPE;
+
+      const preConsentStorageType = getStorageTypeFromPreConsent(state, sessionKey);
+
+      // Storage type precedence order: pre-consent strategy > entry type > global type > default
+      const storageType =
+        preConsentStorageType ?? providedStorageType ?? globalStorageType ?? DEFAULT_STORAGE_TYPE;
```

Comment:
Post consent, there will be another storageType preference before preconsent. Am I correct?

Reply:
Yes, it'll take precedence.